test/cli-rc.test.js

```diff
@@ -169,86 +198,101 @@ test('cli', (t) => {
     const env = { TERM: ' dumb', TZ: 'UTC' }
     const child = spawn(process.argv[0], [bin], { env, cwd: tmpDir })
     // Validate that the time has been translated and correct message key has been used
-    child.on('error', t.threw)
-    child.stdout.on('data', (data) => {
-      t.equal(data.toString(), '[21:31:36.006] FATAL: There was an error starting the process.\n    QueryError: Error during sql query: syntax error at or near SELECTT\n        at /home/me/projects/example/sql.js\n        at /home/me/projects/example/index.js\n    querySql: SELECTT * FROM "test" WHERE id = $1;\n    queryArgs: 12\n')
+    child.on('error', t.assert.fail)
+    const endPromise = new Promise(resolve => {
+      child.stdout.on('data', (data) => {
+        t.assert.strictEqual(data.toString(), '[21:31:36.006] FATAL: There was an error starting the process.\n    QueryError: Error during sql query: syntax error at or near SELECTT\n        at /home/me/projects/example/sql.js\n        at /home/me/projects/example/index.js\n    querySql: SELECTT * FROM "test" WHERE id = $1;\n    queryArgs: 12\n')
+        resolve()
+      })
     })
     child.stdin.write('{"level":60,"time":1594416696006,"msg":"There was an error starting the process.","type":"Error","stack":"QueryError: Error during sql query: syntax error at or near SELECTT\\n    at /home/me/projects/example/sql.js\\n    at /home/me/projects/example/index.js","querySql":"SELECTT * FROM \\"test\\" WHERE id = $1;","queryArgs":[12]}\n')
-    t.teardown(() => {
+    await endPromise
+    t.after(() => {
       fs.unlinkSync(configFile)
       child.kill()
     })
   })
 
-  t.test('throws on missing config file', (t) => {
+  await t.test('throws on missing config file', async (t) => {
     t.plan(2)
     const args = [bin, '--config', 'pino-pretty.config.missing.json']
     const env = { TERM: ' dumb', TZ: 'UTC' }
     const child = spawn(process.argv[0], args, { env, cwd: tmpDir })
-    child.on('close', (code) => t.equal(code, 1))
     child.stdout.pipe(process.stdout)
     child.stderr.setEncoding('utf8')
     let data = ''
     child.stderr.on('data', (chunk) => {
       data += chunk
     })
-    child.on('close', function () {
-      t.match(
-        data.toString(), 'Error: Failed to load runtime configuration file: pino-pretty.config.missing.json')
+    await new Promise(resolve => {
+      child.on('close', function (code) {
+        t.assert.strictEqual(code, 1)
+        match(
+          data.toString(), 'Error: Failed to load runtime configuration file: pino-pretty.config.missing.json', t)
+        resolve()
+      })
     })
```

Comment:
To avoid double promises, I put everything in the same one, but let me know if you prefer to use two promises.